WordPress/Classes/Utility/Version.swift

```diff
@@ -0,0 +1,139 @@
+/// A version according to the semantic versioning specification.
+///
+public struct Version: Sendable {
+
+    /// The major version according to the semantic versioning standard.
+    public let major: Int
+
+    /// The minor version according to the semantic versioning standard.
+    public let minor: Int
+
+    /// The patch version according to the semantic versioning standard.
+    public let patch: Int
+
+    /// The pre-release identifier according to the semantic versioning standard, such as `-beta.1`.
+    public let prereleaseIdentifiers: [String]
+
+    /// The build metadata of this version according to the semantic versioning standard, such as a commit hash.
+    public let buildMetadataIdentifiers: [String]
+
+    /// Initializes a version struct with the provided components of a semantic version.
+    ///
+    /// - Parameters:
+    ///   - major: The major version number.
+    ///   - minor: The minor version number.
+    ///   - patch: The patch version number.
+    ///   - prereleaseIdentifiers: The pre-release identifier.
+    ///   - buildMetaDataIdentifiers: Build metadata that identifies a build.
+    ///
+    /// - Precondition: `major >= 0 && minor >= 0 && patch >= 0`.
+    /// - Precondition: `prereleaseIdentifiers` can contain only ASCII alpha-numeric characters and "-".
+    /// - Precondition: `buildMetaDataIdentifiers` can contain only ASCII alpha-numeric characters and "-".
+    public init(
+        _ major: Int,
+        _ minor: Int,
+        _ patch: Int,
+        prereleaseIdentifiers: [String] = [],
+        buildMetadataIdentifiers: [String] = []
+    ) {
+        precondition(major >= 0 && minor >= 0 && patch >= 0, "Negative versioning is invalid.")
```

Comment:
I would suggest replacing `precondition` (crashes in prod) with safe failures and `wpAssertionFailure`.

Reply:
Fixed! [abb2724](https://github.com/wordpress-mobile/WordPress-iOS/pull/23224/commits/abb27246f3ebca40adeaecf46e3b39fbbf5c250a)